### DISCUSSION TITLE
import screenshot with time in filename

### DISCUSSION THREAD

**Discussion by MKwareContributions** [ImageMagick/ImageMagick#4132] [Help] [answered]
I want to make screenshot with time in it's filename. Something like

`screenshot-3:00-01/01/1970.png`

How do I make that?

**Comment by snibgo** [comment]
You haven't linked to an image.

**Reply by MKwareContributions** [reply]
What do you mean 'linked to an image'?

**Comment by snibgo** [comment]
IM is a raster image processor, so I thought you wanted the time in the image. But you really want a time in the filename.

That depends heavily on your script language. In Windows, %~tF is the date/time of the file %F. For me, it contains a space and a colon and slashes, so I would change these to underscores.

**Reply by MKwareContributions** [reply]
Alright, I'll change dashes to underscores. I'm using Linux (openSUSE Tumbleweed). That means filename should look like this:

`screenshot_%r_%m_%d_%Y.png`

right?
How do I accomplish that, using only terminal command (what parameters should I use)?

**Comment by snibgo** [comment]
Sorry, I don't know how to get the date/time of a file in Linux.

**Comment by fmw42** [comment]
This is an ImageMagick forum. You need to use a Linux forum. But simply search Google for "bash date time". For example, you will get https://tecadmin.net/get-current-date-and-time-in-bash/ which has the date command for use in a bash terminal window.

**Reply by MKwareContributions** [reply]
I do know about time formatting in Linux, I just want to know how to put timestamp in `import` screenshot's filename, that's all.

**Comment by fmw42** [comment]
`import filename.suffix  `

It is up to you to get the time stamp and add that as a variable or variables to the filename before using that command.

So if you get the date and time from Linux and put it into a bash variable in your terminal. So something like:

```
date_time=$(date +"%Y-%m-%d_%T")

import filename_$date_time.jpg
```

**Reply by MKwareContributions** [reply]
That means, if I want to make this a keybind, I need to make .sh file, and bind a key that will execute this file?

**Reply by MKwareContributions** [reply]
Anyways it didn't worked. Filename is now 
`screenshot_.jpg`
Looks like this variable didn't worked.

**Reply by fmw42** [reply]
What was your code to get the variable and insert it?

**Reply by MKwareContributions** [reply]
`date=$("%r_%m/%d/%Y")

import ~/Pictures/screenshots/screenshot_$date.png`

**Comment by fmw42** [comment]
Sorry, I do not know what a keybind is?

**Reply by MKwareContributions** [reply]
Keybind is command (or set of commands) that linked to pressing of key (or combination of keys) on keyboard.

**Comment by fmw42** [comment, marked as answer]
`date +%m-%d-%Y`
08-31-2021

See https://www.tutorialkart.com/bash-shell-scripting/bash-date-format-options-examples/

So try

```
the_date=$(date +%m-%d-%Y)
import filename_$the_date.jpg
```

**Reply by MKwareContributions** [reply]
Thank you! This worked!
